extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampSearchExtractor.java

```diff
@@ -0,0 +1,123 @@
+// Created by Fynn Godau 2019, licensed GNU GPL version 3 or later
+
+package org.schabi.newpipe.extractor.services.bandcamp.extractors;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.schabi.newpipe.extractor.InfoItem;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.downloader.Downloader;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.linkhandler.SearchQueryHandler;
+import org.schabi.newpipe.extractor.search.InfoItemsSearchCollector;
+import org.schabi.newpipe.extractor.search.SearchExtractor;
+import org.schabi.newpipe.extractor.services.bandcamp.extractors.streaminfoitem.BandcampPlaylistStreamInfoItemExtractor;
+import org.schabi.newpipe.extractor.services.bandcamp.extractors.streaminfoitem.BandcampSearchStreamInfoItemExtractor;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+public class BandcampSearchExtractor extends SearchExtractor {
+
+    public BandcampSearchExtractor(StreamingService service, SearchQueryHandler linkHandler) {
+        super(service, linkHandler);
+    }
+
+    @Override
+    public String getSearchSuggestion() {
+        return null;
+    }
+
+    @Override
+    public boolean isCorrectedSearch() {
+        return false;
+    }
+
+    @Override
+    public InfoItemsPage<InfoItem> getPage(String pageUrl) throws IOException, ExtractionException {
+        // okay apparently this is where we DOWNLOAD the page and then COMMIT its ENTRIES to an INFOITEMPAGE
+        String html = getDownloader().get(pageUrl).responseBody();
+
+        InfoItemsSearchCollector collector = new InfoItemsSearchCollector(getServiceId());
+
+
+        Document d = Jsoup.parse(html);
+
+        Elements searchResultsElements = d.getElementsByClass("searchresult");
+
+        for (Element searchResult :
+                searchResultsElements) {
+
+            String type = searchResult.getElementsByClass("result-info").first()
+                    .getElementsByClass("itemtype").first().text();
+
+            switch (type) {
+                default:
+                    continue;
+                case "FAN":
+                    // don't display fan results
```

Comment:
why? isn't it because #279 is not yet implemented?

Reply:
I wrote above:

>Fan pages  
>Are extra work because they look different and don't really provide a lot of value I think

A fan page is a collection of items that a user (fan) has bought. It doesn't really make sense to implement them to me, and it might mislead users to think that a fan is the creator of the items in their account.